IC-898: Add service provider dashboard

lawrence-forooghian opened the pull request:
## What does this pull request do?

This adds the first version of a page where a service provider can see all of the referrals that have been sent by probation practitioners using the service.

## What is the intent behind these changes?

To demonstrate that a service provider can see a referral after a probation practitioner submits it.

## Screenshot

![Receive dashboard -- User views a list of sent referrals](https://user-images.githubusercontent.com/53756884/105862932-433c2a80-5fe8-11eb-87b4-090cfb84a0ac.png)

Gweaton commented on server/utils/calendarDay.test.ts:
```diff
@@ -60,4 +60,11 @@ describe('CalendarDay', () => {
   })
+
+  describe('anyUTCMoment', () => {
+    it('returns a Date which, in the UTC time zone, lies within the day represented by the receiver', () => {
```
I'm finding this a little hard to understand - probably just me though, would you mind explaining what this is doing?

lawrence-forooghian replied:
Haha, yeah, I'll have a think about how to word it more clearly.

lawrence-forooghian commented:
@tomsmyers I've realised that I've added a contract test as part of a functionality PR - want me to split it out or is this an uncontroversial one?

Gweaton commented on server/routes/receive/dashboardPresenter.ts:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    const day = CalendarDay.parseIso8601(referral.sentAt)
+    let formattedDay = null
```
Not a blocker, but this might be a bit cleaner as: `const formattedDay = day ? ReferralDataPresenterUtils.govukShortFormattedDate(day) : ''`

lawrence-forooghian replied:
Yeah, I think that's nicer, will change it.

Gweaton commented on server/routes/receive/receiveController.ts:
```diff
@@ -0,0 +1,24 @@
+import DashboardPresenter from './dashboardPresenter'
+import DashboardView from './dashboardView'
+
+export default class ReceiveController {
```
Feel free to ignore this pedantic comment, but now that we've renamed the service to "Find, arrange and monitor", is there another name that might be clearer to future readers of the codebase if `Receive` is no longer a name? Only ones I can think of are `ServiceProviderController` (but I feel the controller should reflect the entity it uses the most (would be a `referral` and we already have a controller for that...). `ServiceProviderReferralsController`? 

I was about to type `arrangeController` but that also applies to "Make a referral", so this is hard...

lawrence-forooghian replied:
Yeah, that's a great point. I think that `ServiceProviderReferralsController` is good. I'll change it.

Gweaton reviewed (approved):
Well done on this, looks like a lot of work! A couple of comments, but nothing major.

lawrence-forooghian commented:
Looks like @tomsmyers introduced a "full name" in a presenter in #87 too, will update it to use what I've added here.

lawrence-forooghian commented:
> Looks like @tomsmyers introduced a "full name" in a presenter in #87 too, will update it to use what I've added here.

Same for date formatting, too, updating that.

lawrence-forooghian commented:
@Gweaton I addressed your comments. I also realised that the current code didn't handle British Summer Time, so I've added some timezone handling stuff to `CalendarDay`. Would you mind taking a look that that class makes sense?
